integration/experiment/system_characterization/gen_plot_cpu_heatmap.py

```diff
@@ -0,0 +1,191 @@
+            value = value[metric].mean()
+            temp.append(value)
+        zs.append(temp)
+    if show_details:
+        summary = pandas.DataFrame(zs, columns=ys, index=xs)
+        summary.index.name = 'core_frequency'
+        sys.stdout.write('{}\n{}\n'.format(metric, summary))
+    return xs, ys, zs, min_z, max_z
+
+
+def plot_uncore_sweep_heatmap(df, metric, label, region, output_dir, show_details):
+    if metric not in ['power', 'energy', 'runtime']:
+        raise RuntimeError('Unknown z-axis metric: {}'.format(metric))
+
+    xs, ys, zs, min_z, max_z = setup_3d_data(df, metric, show_details)
+
+    if (metric != 'runtime'):
+       min_z = round(min_z-10, -1)
+       max_z = round(max_z+10, +1)
+
+    value_range = np.linspace(min_z, max_z, 10)
+    z_thresh = 0.0
+    cmap = cm.get_cmap('rainbow')
+    norm = colors.Normalize(value_range.min(), value_range.max())
+
+    metric_label = { 'runtime': 'Runtime (sec)',
+                     'energy': 'Energy (j)',
+                     'power': 'Power (W)'}
+
+    plot_heatmap(data=zs, cmap=cmap, norm=norm, zbar_range=value_range,
+                 zbar_label=metric_label[metric], z_thresh=z_thresh,
+                 x_range=xs, x_label='Core Frequency (GHz)',
+                 y_range=ys, y_label='Uncore Frequency (GHz)',
+                 title='{} Core-Uncore Frequency Sweep\nregion = {}, z-axis = {}'.format(label, region, metric),
+                 outdir=output_dir,
+                 filename='{}_uncore_frequency_sweep_{}_{}'.format(label.lower(), metric, region.lower()))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    common_args.add_output_dir(parser)
+    common_args.add_show_details(parser)
+    common_args.add_label(parser)
+    parser.add_argument('--target-region', dest='target_region',
+                        action='store', type=str, default=None,
+                        help='name of the region to use to select data (default=application totals)')
+    parser.add_argument('--metric', dest='metric',
+                        action='store', type=str, default='power',
+                        help='metric to use for z-axis, one of power, energy, or runtime (default: power)')
+    parser.add_argument('--plot-3d', dest='plot_3d',
+                        action='store_true', default=False,
+                        help='plot as a 3D surface instead of a heatmap')
+    parser.add_argument('--hostname', required=True, dest='hostname',
+                        action='store', default=None,
+                        help='Hostname of the node to which the report belongs')
+    parser.add_argument('--boardcap', required=True, dest='boardcap',
+                        action='store', default=None,
+                        help='Platform-level board power cap')
+
+    args = parser.parse_args()
+    output_dir = args.output_dir
+    regex_report_format = f"*boardcap_{args.boardcap}_*_trial_1_*{args.hostname}*report"
```

Comment:
```suggestion
    regex_report_format = f"*boardcap_{args.boardcap}_*_trial_*_*{args.hostname}*report"
```
Just noticed the trial was hardcoded to 1.  Was that intentional?

Reply:
Yes. This was intentional. This script refers to generating the plots - so we only need data from a single trial.